### ISSUE TITLE
python3 library

### ISSUE THREAD

**Issue by mahaase** [BobBuildTool/basement#168]
hey,

lovely to see the python3 support in bob basement https://github.com/BobBuildTool/basement/commit/ef077d2b4c1f0c26802faee0d6c842f29257ce73.

but how could we handle, if a package needs python3 library and header files?
this could not be realized with the current https://github.com/BobBuildTool/basement/blob/master/recipes/python/python3.yaml recipe.

do u have something in mind, how to support this?

in my first tests i realized, that i makes no sense to build a shared python3-dev package for target-toolchain.
in the end python3 tries to use some libraries while buildtime, which cant be resolved then. for host-toolchain, the python3 package will be implicitly static too.

so i guess we could extend the python3 recipe with:
```
privateEnvironment:
    BASEMENT_LIBS: "static"

[...]
        multiPackage:
            "":
                packageScript: |
                    autotoolsPackageTgt
                    ln -sf python3 usr/bin/python

            dev:
                packageScript: |
                    rsync -a "$1/install/" .
                    ln -sf python3 usr/bin/python

        provideTools:
            python3: usr/bin
```

and because static is forced, a tgt package makes no sense?!

**UPDATE**: that part with static lib, is wrong. what i tried to say is, that we may need a shared python3 library without dependencies to other shared libraries. so the dependencies have to be static, not python3 itself.

maybe something like this:
```
depends:
  - environment:
      BASEMENT_LIBS: "static"
    depends:
      - libs....
```

**Comment by jkloetzke** [comment]
The most important question is for what environment is the python library intended? Does it need to be "cross-compiled" and runs with a cross-compiled pyhton3 on a target system? Or is the package intended to be used with the host "python3" tool?

Also, do you know a simple example library that needs the python3 library/headers for testing?

**Comment by mahaase** [comment]
i checked some projects more deeply and realized, that in all cases so far, python was just needed for the host system. i have no idea, if this would also be the case for usage of pybind11 stuff. but for current use-cases i will close this ticket and open an new one, if problem is in focus again. thanks so far.

**Comment by mahaase** [comment]
> now there is a `python3-dev` package (by #182), but what is the idea to get the correct paths to python include directory and to python3 library?
> 
> ofc i could to some magic to search for `Python3.h` to get the include path, but for the library that's a bit harder and everybody has to do this, who want's to use that package. i guess we need a better solution here.
> 
> btw: and if we have a `python3-dev` package, we also will need a `python3-tgt` package. if i use the package with `target-toolchain` i will have a shared library, that has to be installed.

for now, I just found this solution, which is pretty stupid, if `python3.8` isn't used 🙄

```
export PYTHONLIB="$(find ${BOB_DEP_PATHS[python::python3-dev]}/usr/lib -type f -xtype f -name 'libpython3.8*')"
export PYTHONINC="$(dirname `find ${BOB_DEP_PATHS[python::python3-dev]}/usr/include -name Python.h`)"
```

**Comment by jkloetzke** [comment]
Is this issue still open?  I think this was solved by d4584321b71b48ea134b647fdb6f6ece25fca26a, wasn't it?

**Comment by mahaase** [comment]
indeed! works very well now. i was able to remove all workarounds, that i did in our recipes! 
well done & thx alot!
